src/test/java/htsjdk/samtools/util/EdgeReadIteratorTest.java

```diff
@@ -333,6 +336,21 @@ public void testIntersectingAndNotInterval() {
         assertEquals(81, locusPosition);
     }
 
+    @Test
+    public void testNoGapsInLocusAccumulator() {
+        final SamReader reader = SamReaderFactory.make().open(new File("src/test/resources/htsjdk/samtools/util/sliver.sam"));
+        final EdgeReadIterator iterator = new EdgeReadIterator(reader, null);
+
+        AbstractLocusInfo<EdgingRecordAndOffset> previous=null;
+        while (iterator.hasNext() && (previous == null || previous.getPosition() < 1_000_000)) {
+            final AbstractLocusInfo<EdgingRecordAndOffset> next = iterator.next();
+            if (previous != null) {
+                Assert.assertEquals(next.getPosition(), previous.getPosition() + 1);
+            }
+            previous = next;
+        }
```

Comment:
It's probably not worth fixing, but note that this test will pass even if the `while` loop never loops (or only loops once). If you know how many iterations are expected, you could count them and assert that at the end too.